service/src/NVMLIOGroup.cpp

```diff
@@ -258,6 +268,17 @@ namespace geopm
             // to the min & max supported frequency for the associated signals
             m_frequency_max_control_request.at(domain_idx) = m_supported_freq.at(domain_idx).back() * 1e6;
             m_frequency_min_control_request.at(domain_idx) = m_supported_freq.at(domain_idx).front() * 1e6;
+
+            if (m_supported_freq.at(domain_idx).size() >= 2) {
```

Comment:
`m_supported_freq.at(domain_idx)` is equivalent to `supported_frequency` in this loop, right? If so, maybe use `supported_frequency` in this check, since that's the variable we're using as a dependency on the check.